geth/jail/cell.go

```diff
@@ -65,9 +67,28 @@ func (c *Cell) Stop() {
 // event queue loop and schedules for immediate execution.
 // Intended to be used by any cell user that want's to run
 // async call, like callback.
-func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) {
+func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) error {
 	task := looptask.NewCallTask(fn, args...)
-	c.lo.Add(task)
-	// TODO(divan): review API of `loop` package, it's contrintuitive
-	go c.lo.Ready(task)
+	ch1 := make(chan error, 1)
+	ch2 := make(chan int)
```

Comment:
Here the typical way is to use a channel with an empty struct and call it e.g. `doneC`, depending on the intention. The reader should unterstand why it is created.

Simple way w/o error handling:

```
doneC := make(chan struc{})

go func() {
    ...
    doneC <- struct{}{}
}()

...

<-doneC
```